### PR TITLE
Fix Grammar.__repr__ backslash escaping

### PR DESCRIPTION
**Pull request by ingolemo**
Fixes the backslash escaping by making use of python's own string representations.

It has the side-effect that `'` will usually be preferred over `"` when printing literals. This affects several tests, which had to be updated.

### CODE REVIEW

**Review by evanh — COMMENTED** [review]
I would add an explicit test case for this. The changes look good, thanks for this!

**Review by evanh — APPROVED** [review]
LGTM, thanks!

File: parsimonious/tests/test_expressions.py [hunk 1]
```diff
@@ -289,7 +289,27 @@ def test_unicode_surrounding_parens(self):
...
+        eq_(repr(Grammar(r'''foo = "\\" ''')), u'''Grammar("foo = '\\\\\\\\'")''')
+
+        # single quote
+        eq_(repr(Grammar(r'''foo = "'" ''')), u'''Grammar('foo = "\\'"')''')
+
+        # escaped single quote inside a single quoted string
+        eq_(repr(Grammar(r'''foo = '\'' ''')), u'''Grammar('foo = "\\'"')''')
+
+        # double quote
+        eq_(repr(Grammar(r'''foo = '"' ''')), u'''Grammar('foo = \\'"\\'')''')
```
Comment (erikrose) [inline parsimonious/tests/test_expressions.py:309]: The ones of these I've spot checked appear right, but boy are they mind-bending! We could prove they are right, without doing much extra work, by round-tripping them in the test and making sure the results are equal. For example…

```python
g = Grammar(...)
eq_(g, eval(repr(g)))
```

I believe all the `__eq__` methods I've defined in expressions.py should do the heavy lifting. No need to block this PR on this, though—just something to think about for later.

**Comment by erikrose** [comment]
Thanks, @ingolemo, for the patch and @evanh for the review! :-D

**Pull request merged**